tests/miral/runner.cpp

```diff
@@ -80,15 +55,15 @@ TEST_F(Runner, stop_callback_is_called)
 
 TEST_F(Runner, start_callback_is_called)
 {
-    mt::Signal signal;
+    auto signal = std::make_shared<mt::Signal>();
```

Comment:
Why introduce a `shared_ptr`?

I do realise that in the failure case the lambda will have a stale reference to `signal`, and that _could_ lead to undefined behaviour (if the reference were actually used).

However, taking the `shared_ptr` by reference doesn't change that.